### PR TITLE
Features/31802 text field counter length support

### PR DESCRIPTION
**Pull request by kimhjona**

### CODE REVIEW

File: example/src/App.tsx [hunk 1]
```diff
@@ -373,13 +383,14 @@ class App extends React.Component<{}, State> {
...
         <Heading>Connected Text Field</Heading>
+        <Loading />
         <TextField
           label="Connected Text Field"
           type="number"
           placeholder=""
           value={this.state.appTextCounter}
           helpText="Helper Text"
-          enableTextCouter={true}
+          /* enableTextCouter={true} */
```
Comment (mkulinski) [inline example/src/App.tsx]: comment?

File: src/components/TextField/TextField.scss [hunk 2]
```diff
@@ -7,6 +7,10 @@ $stacking-order: (
   backdrop: 10,
 );
 
+.red {
+  color: red;
```
Comment (mkulinski) [inline src/components/TextField/TextField.scss]: is there a specific red we should make this? (Todd question)

Comment (kimhjona) [reply src/components/TextField/TextField.scss]: screen shot 2017-09-27 at 1 08 22 pm
todd asked for red, but my guess is that the exact shade would matter too much in this case

**Comment by mkulinski** [comment]
Please resolve merge conflicts and push changes

**Pull request merged**